hare/pb/hare.proto

```diff
@@ -0,0 +1,39 @@
+syntax = "proto3";
+
+package pb;
+option go_package = "pb";
+
+// top message of the protocol
+message HareMessage {
+    bytes pubKey = 1;
+
+    bytes inner_sig = 2; // sign type, layer and payload
+    uint32 type = 3; // message type (status/commit/proposal/notify)
+    bytes layer = 4;
+    BasicMessage message = 5;
+
+    bytes outer_sig = 6; // optional. sign also certificate
+    Certificate cert = 7; // optional
+}
+
+// the certificate
+message Certificate {
+    repeated bytes blockId = 1; // the committed set S
+    repeated HareMessage commits = 2; // a collection of commit messages
```

Comment:
if we remove S and sig from the commits and we stay with public key only. we can just replace repeated HareMessage with repeated public key.

Reply:
I think the other values are not necessarily equal on all message. But I'll keep that point in mind. Anyway, the high overhead comes from attaching S to each message (which is already extracted).